src/test/kotlin/org/rust/ide/actions/mover/RsStatementUpDownMoverTest.kt

```diff
@@ -620,4 +620,27 @@ class RsStatementUpDownMoverTest : RsStatementUpDownMoverTestBase() {
              };
         }
     """, testmark = UpDownMoverTestMarks.moveOutOfBody)
+
+    // https://github.com/intellij-rust/intellij-rust/issues/7376
+    fun `test move down in block inside argument list`() = moveDown("""
+        fn foo(a: u32) {}
+
+        fn main() {
+            foo({
+                println!("1");/*caret*/
```

Comment:
What about moving up in this case?

Looks like the current implementation does not check boundaries and might produce invalid code, for example
moving up here
```Rust
foo(
    {
        println!("2"); // move up
        println!("1");
        1
    }
);
```
leads to
```Rust
fn main1() {
    foo(
        println!("2");
    {
            println!("1");
            1
        }
    );
}
```

Reply:
Good catch! The only comma separated lists that accept expressions are argument lists (AFAIK), so I added a check for them. It should be fixed now.

Reply:
Great, now the implementation works in that case. But I found another problematic case
```Rust
S {
    a: 42,
    b: foo(|| {
        bar; // move up
    })
};
```
Could you please try to fix it as well and add the corresponding tests?

Reply:
Thanks, fixed.

Reply:
Well, how's about this? 🙂 
```Rust
S {
    a: 42,
    b: foo(&mut || {
        bar; // move up
    })
};
```

You can find such problems too by manually moving some random lines inside block expressions in some real projects (e.g. `testData/rust-analyzer`). BTW probably such fuzz testing could be performed automatically someday

Reply:
Ok I guess there's no need to be prudent, now any block expr inside argument lists will be considered. Maybe the heuristic could be improved (stop at blocks?), but it seems to me that this is such a rare edge case that it's not worth it.

Reply:
Yes, I think it's mostly fine now 👍